app/src/main/java/com/example/tasky/common/presentation/TextBox.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.tasky.common.presentation
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun TextBox(hintText: String) {
+    var text by remember { mutableStateOf("") }
```

Comment:
Keeping such text state in the composable itself is very dangerous, since this won't survive configuration changes like screen rotations. It also won't survive process death (video about that is in the video portal of this program).

You could bypass both by using `rememberSaveable`, but even that I wouldn't recommend, since the more complex the app becomes the more messy the Compose UI code will become. That's why we typically let our ViewModels manage hosting and updating our state. That not only provides a better separation of concerns, but also lets you test this logic easier since you don't need to spin up a UI test for that.